src/main.rs

```diff
@@ -41,20 +41,40 @@ async fn main() -> anyhow::Result<()> {
             .map(|streamer_message| handle_streamer_message(streamer_message, &pool))
             .buffer_unordered(1usize);
 
-        while let Some(_handle_message) = handlers.next().await {}
+        let mut time_now = std::time::Instant::now();
+        while let Some(handle_message) = handlers.next().await {
+            match handle_message {
+                Ok(block_height) => {
+                    let elapsed = time_now.elapsed();
+                    tracing::info!(
+                        target: LOGGING_PREFIX,
+                        "Elapsed time spent on block {}: {:.3?}",
+                        block_height,
+                        elapsed
+                    );
+                    time_now = std::time::Instant::now();
+                }
+                Err(e) => {
+                    tracing::error!(target: LOGGING_PREFIX, "Stop indexing due to {}", e);
+                    // we do not catch this error anywhere, this thread is just stopped with error,
+                    // main thread continues serving metrics
+                    anyhow::bail!(e)
+                }
+            }
+        }
+        Ok(()) // unreachable statement, loop above is endless
     });
 
-    metrics::init_metrics_server().await?;
-
-    Ok(())
+    metrics::init_metrics_server(opts.port).await
 }
 
 async fn handle_streamer_message(
     streamer_message: near_indexer_primitives::StreamerMessage,
     pool: &sqlx::Pool<sqlx::Postgres>,
 ) -> anyhow::Result<u64> {
     metrics::BLOCK_PROCESSED_TOTAL.inc();
-    // Prometheus Guage Metric type only supports i64 and f64 types for the time being, so we need to cast the type into i64
+    // Prometheus Gauge Metric type do not support u64
+    // https://github.com/tikv/rust-prometheus/issues/470
```

Comment:
@roshaans that's what I meant when I asked to add the link to the issue in the comments